cpp/include/cudf_test/column_wrapper.hpp

```diff
@@ -363,6 +432,33 @@ class fixed_width_column_wrapper : public detail::column_wrapper {
                                    cudf::UNKNOWN_NULL_COUNT});
   }
 
+  template <typename InputIterator, typename ValidityIterator>
+  fixed_width_column_wrapper(thrust::input_host_iterator_tag,
+                             InputIterator begin,
+                             InputIterator end,
```

Comment:
I'd rather see us inferring the iterator tag type from the iterator and dispatching accordingly.

Reply:
I tried `copy_n` which should figure out the system automatically. (host/device/cross_system) 
It works for this benchmark but fails to compile for TESTS and other benchmarks.

For example, huge number of errors like
```
In file included from tmpxft_00006ad6_00000000-6_concatenate_tests.cudafe1.stub.c:1:
/tmp/tmpxft_00006ad6_00000000-6_concatenate_tests.cudafe1.stub.c:66:393: error: template argument 1 is invalid
   66 | typedef thrust::cuda_cub::__parallel_for::ParallelForAgent< ::thrust::cuda_cub::__transform::unary_transform_f< ::thrust::transform_iterator< ::cudf::test::detail::fixed_width_type_converter<int, int> ,  ::thrust::transform_iterator< ::,  ::thrust::counting_iterator<int,  ::thrust::use_default,  ::thrust::use_default,  ::thrust::use_default> ,  ::thrust::use_default,  ::thrust::use_default> ,  ::thrust::use_default,  ::thrust::use_default> ,  ::thrust::device_ptr<int> ,  ::thrust::cuda_cub::__transform::no_stencil_tag,  ::thrust::identity<int> ,  ::thrust::cuda_cub::__transform::always_true_predicate> , long>  _ZN6thrust8cuda_cub14__parallel_for16ParallelForAgentINS0_11__transform17unary_transform_fINS_18transform_iteratorIN4cudf4test6detail26fixed_width_type_converterIiiEENS5_IZN27OverflowTest_Presliced_Test8TestBodyEvEUliE_NS_17counting_iteratorIiNS_11use_defaultESE_SE_EESE_SE_EESE_SE_EENS_10device_ptrIiEENS3_14no_stencil_tagENS_8identityIiEENS3_21always_true_predicateEEElEE;
      |                                                                                                                                                                                                                                                                                                                                                                                                         ^
/tmp/tmpxft_00006ad6_00000000-6_concatenate_tests.cudafe1.stub.c:66:443: error: template argument 2 is invalid
   66 | typedef thrust::cuda_cub::__parallel_for::ParallelForAgent< ::thrust::cuda_cub::__transform::unary_transform_f< ::thrust::transform_iterator< ::cudf::test::detail::fixed_width_type_converter<int, int> ,  ::thrust::transform_iterator< ::,  ::thrust::counting_iterator<int,  ::thrust::use_default,  ::thrust::use_default,  ::thrust::use_default> ,  ::thrust::use_default,  ::thrust::use_default> ,  ::thrust::use_default,  ::thrust::use_default> ,  ::thrust::device_ptr<int> ,  ::thrust::cuda_cub::__transform::no_stencil_tag,  ::thrust::identity<int> ,  ::thrust::cuda_cub::__transform::always_true_predicate> , long>  _ZN6thrust8cuda_cub14__parallel_for16ParallelForAgentINS0_11__transform17unary_transform_fINS_18transform_iteratorIN4cudf4test6detail26fixed_width_type_converterIiiEENS5_IZN27OverflowTest_Presliced_Test8TestBodyEvEUliE_NS_17counting_iteratorIiNS_11use_defaultESE_SE_EESE_SE_EESE_SE_EENS_10device_ptrIiEENS3_14no_stencil_tagENS_8identityIiEENS3_21always_true_predicateEEElEE;
      |                                                                                                                                                                                                                                                                                                                                                                                                                                                           ^                                                                                                                                                                                             
```

ninja -j 12 tests/CMakeFiles/COPYING_TEST.dir/copying/concatenate_tests.cu.o

Reply:
Yes. It deduces wrong space.
Iterator category is `thrust::detail::iterator_category_with_system_and_traversal<std::random_access_iterator_tag, thrust::any_system_tag, thrust::random_access_traversal_tag>`
system used is `thrust::any_system_tag`. Not sure what it does at this point.

```c++
using Category = typename thrust::iterator_traits<InputIterator>::iterator_category;
thrust::detail::is_device_iterator_category<Category>::value;
thrust::detail::is_host_iterator_category<Category>::value;
```

I tried a tagged iterator with thrust::input_device_iterator_tag or thrust::random_access_device_iterator_tag
`thrust::detail::iterator_category_with_system_and_traversal<std::random_access_iterator_tag, thrust::input_device_iterator_tag, thrust::random_access_traversal_tag>`
Even then, is_device_iterator_category==false and is_host_iterator_category==true. 
I don't know why it does not work as expected.

Reply:
@karthikeyann Is this resolved? It looked like you fixed it.

Reply:
Yes. Tagged with `thrust::device_system_tag` and it works now.